crates/circuit/src/dag_circuit.rs

```diff
@@ -5631,16 +5635,47 @@ impl DAGCircuit {
         Ok(nodes.into_iter())
     }
 
+    fn reversible_topological_nodes(
```

Comment:
I would avoid adding this function and the `reversible_topological_op_nodes` function and just go with adding the `reverse` parameter to the existing `topological_nodes` and `topological_op_nodes` functions.

Reply:
I agree there is a tradeoff here, but I'd lean towards eliminating code duplication in this case. Especially since the `reversible_topological_*` functions called with `reverse` set to `false` is essentially doing the same thing as just calling the corresponding `topological_*` functions. FWIW, there are several functions which would certainly merit default/optional parameters but instead we call with the same signature most of the time, for example with `DAGCircuit::from_circuit_data` and `CircuitData::copy_empty_like`.